mangaki/mangaki/utils/recommendations.py

```diff
@@ -48,13 +48,12 @@ def get_reco_algo(user, algo_name='knn', category='all'):
     chrono.save('get all %d interesting ratings' % queryset.count())
 
     dataset = Dataset()
-    backup_filename = '%s.pickle' % algo_name
-    if os.path.isfile(os.path.join('pickles', backup_filename)):  # When Algo class will be there: 'if algo.has_backup():'
-        algo = ALGOS[algo_name]()
-        algo.load(backup_filename)
-        dataset.load('ratings-' + backup_filename)
+    algo = ALGOS[algo_name]()
+    if algo.has_backup():
+        algo.load(algo.get_backup_filename())
```

Comment:
If it's frequent to use the default filename, consider use it as a default argument for the load method.